source/Tests/MessageContractAnalyzerFixture.cs

```diff
@@ -276,78 +276,100 @@ public class SimpleResponse : IResponse { }
         }
 
         [Test]
-        public async Task OptionalPropertiesOnMessageTypesMustBeNullable() // except collection types
+        public async Task OptionalPropertiesOnMessageTypesMustBeInitializedOrNullable()
         {
             var source = WithOctopusTypes(@"
 namespace Octopus.Core.Features.ServerTasks.MessageContracts
 {
     /// <summary>request</summary>
     public class SimpleRequest: IRequest<SimpleRequest, SimpleResponse> 
     {
+        public SimpleRequest(string requiredString, int requiredInt, bool requiredBool, string[] requiredCollection)
+        {
+            RequiredStringProperty = requiredString;
+            RequiredIntProperty = requiredInt;
+            RequiredBoolProperty = requiredBool;
+            RequiredCollectionProperty = requiredCollection;
+
+            IntProperty = 27; // We don't decompile the constructor, it must be an inline initializer so this doesn't count
+        }
+
+        // --- sanity check required properties aren't affected -----
+
+        [Required]
+        public string RequiredStringProperty { get; set; }
+
+        [Required]
+        public int RequiredIntProperty { get; set; }
+
+        [Required]
+        public bool RequiredBoolProperty { get; set; }
+
+        [Required]
+        public string[] RequiredCollectionProperty { get; set; }
+
+        // --- important stuff starts here -----
+
         [Optional]
-        public string {|#0:StringProperty|} { get; set; }
+        public string {|#0:StringProperty|} { get; set; } // NOT OK, must be nullable or initialized
 
         [Optional]
-        public int {|#1:IntProperty|} { get; set; }
+        public string? NullableStringProperty { get; set; } // OK
 
         [Optional]
-        public string[] StringListProperty { get; set; } = null; // should NOT fire on this because optional nonnull collections are allowed
+        public string InitializedStringProperty { get; set; } =  ""foobar""; // OK
 
         [Optional]
-        public bool BoolProperty { get; set; } // should NOT fire on this because we have a special case for bool
-    }
-    /// <summary>response</summary>
-    public class SimpleResponse : IResponse { }
-}");
+        public string InitializedStringPropertyEmpty { get; set; } = string.Empty; // OK
 
-            await Verify.VerifyAnalyzerAsync(source,
-                new DiagnosticResult(Descriptors.OptionalPropertiesOnMessageTypesMustBeNullable).WithLocation(0).WithArguments("StringProperty", "string"),
-                new DiagnosticResult(Descriptors.OptionalPropertiesOnMessageTypesMustBeNullable).WithLocation(1).WithArguments("IntProperty", "int"));
-        }
+        [Optional]
+        public string InitializedStringPropertyExplicitNull { get; set; } = null!; // OK; If you really want to break things, then you can.
```

Comment:
I'm not sure if this is a great idea... I wonder if we should take a stand against it?

Reply:
Yeah... It requires more work in the analyzer to pick this up, which could also be defeated with something like this

```csharp
static string GetNullString() => null!;
public string InitializedStringPropertyExplicitNull { get; set; } = GetNullString(); 
```

I thought about it, and came to the conclusion that at some point we can't enforce everything and need to rely on code review to pick up bad behaviour; and a code review is highly like to flag an `= null!;` anyway.

Would be good to gather more opinions though if possible.